lib/datadog/core/transport/http/builder.rb

```diff
@@ -70,7 +70,7 @@ def initialize
 
           def adapter(config, *args, **kwargs)
             @default_adapter = case config
-                               when Core::Configuration::AgentSettingsResolver::AgentSettings
+                               when Core::Configuration::AgentSettingsResolverCommon::BaseAgentSettings
```

Comment:
I don't love that we directly reference `BaseAgentSettings` outside of the resolvers, but since we don't know which resolver generated AgentSettings here, I'm not sure of a better way to maintain existing functionality.